### ISSUE TITLE
[BUG] Issue with whitelist

### ISSUE THREAD

**Issue by soerenmartius** [francoismassart/eslint-plugin-tailwindcss#237]
**Describe the bug**
Seems like there's an issue with the whitelisting classes that include colons.
For example, let's consider the following class `i-heroicons:chevron-up-solid`. The following configuration should allow the class to pass. 

```
<span class="i-heroicons:chevron-up-solid w-6 h-6" />
```

```
  settings: {
    tailwindcss: {
      // Testing multiple options
      whitelist: [
        'i\\-.+',
        'i-heroicons:chevron-up-solid',
        'i\\-heroicons:chevron\\-up\\-solid',
      ],
    },
```

However that doesn't work!, also, If I am trying to whitelist exactly the same class instead of using regex the whitelisting will fail. This can be tested by adding the scenario to https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/ea167fd9cb56b7fd1430b93e960736546bf71bd2/tests/lib/rules/no-custom-classname.js#L414

**Comment by soerenmartius** [comment]
Seems like classes that include a colon are problematic.

**Comment by francoismassart** [comment]
Hi,
that's because the plugin only parse the last part of classNames (after the last `:`)..
I'll see if I what fix it fast

**Comment by francoismassart** [comment]
@soerenmartius 👉  `npm i eslint-plugin-tailwindcss@3.11.1-beta.0`

**Comment by soerenmartius** [comment]
> @soerenmartius 👉 `npm i eslint-plugin-tailwindcss@3.11.1-beta.0`

thanks for your prompt reply - just tested the new beta but unfortunately it seems like the issue remains.

```
  settings: {
    tailwindcss: {
      whitelist: [
        'i\\-.+', // <-- this is still faliling
        '.+-astroblack',
      ],
    },
  },

```

**Comment by soerenmartius** [comment]
> @soerenmartius What are you using as lib for integrating icon with the prefix `i-heroicons:chevron-up-solid:` ?

[UnoCSS](https://github.com/unocss/unocss) and its [icons preset](https://unocss.dev/presets/icons). It's really neat!

**Comment by soerenmartius** [comment]
```
    {
      code: `<div className="flex skin-summer custom-2 i-heroicons:chevron-up-solid", "x-heroicons:chevron-down-solid">whitelisted</div>`,
      options: [
        {
          whitelist: ["skin\\-(summer|xmas)", "custom\\-[1-3]", "i\\-.+", "x-heroicons:chevron-down-solid"],
        },
      ],
    },
```

I'd probably add  test scenario for exact matching as well as regex matching as shown in my example.

**Comment by francoismassart** [comment]
@soerenmartius it's because on this beta the whitelisting of classnames with `:` (or any `separator`) is only ran against whitelisted elements that contains it.

If I don't do it like that I would introduce a side effect blocking the use of negated whitelist like this one:
<img width="872" alt="image" src="https://user-images.githubusercontent.com/704026/231510739-8e752621-dfe1-4756-a620-7fc7675a5ecc.png">

I guess you could do:

```json5
  settings: {
    tailwindcss: {
      whitelist: [
        'i\\-.+:?.+',
        '.+-astroblack',
      ],
    },
  },
```

**Comment by mariusc23** [comment]
I am also running into this issue with colons not being allowed. My use case is adding custom [tailwind variants](https://tailwindcss.com/docs/plugins#adding-variants). The beta version (`3.11.1-beta.0`) fixes it for me!

**Comment by soerenmartius** [comment]
> ```js
>         'i\\-.+:?.+',
> ```

Makes sense @francoismassart  - I am super cool with the solution. Thanks for addressing this quickly, very much appreciated

**Comment by soerenmartius** [comment]
> @soerenmartius, can we close this issue ?

yes awesome thank you
